lib/transports/__es__/_serverless.js

```diff
@@ -1,13 +1,52 @@
 const jsonParser = require('../../jsonparser.js')
 const aws4signer = require('../../aws4signer')
 const _ = require('lodash')
+const { searchAfterResultSet } = require('./_helpers')
 
 class Serverless {
   async getOpenSearchServerlessData (limit, offset, callback) {
     const searchBody = await this.searchWithTemplate(this.searchBody)
     searchBody.size = limit
     searchBody.from = this.currentOffset ? this.currentOffset : offset
+
+    // Use search_after if specified in options
```

Comment:
feels like a duplicate. this can be made into a utility function that is shared by both. _severless and _data


should accept, self, url and necessary configs